sdk/helper/testcluster/util.go

```diff
@@ -92,23 +93,43 @@ func NodeSealed(ctx context.Context, cluster VaultCluster, nodeIdx int) error {
 }
 
 func WaitForNCoresSealed(ctx context.Context, cluster VaultCluster, n int) error {
-	for ctx.Err() == nil {
-		sealed := 0
-		for i := range cluster.Nodes() {
-			ctx, cancel := context.WithTimeout(ctx, 5*time.Millisecond)
-			if err := NodeSealed(ctx, cluster, i); err == nil {
-				sealed++
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	errs := make(chan error)
+	for i := range cluster.Nodes() {
+		go func(i int) {
+			var err error
+			for ctx.Err() == nil {
+				err = NodeSealed(ctx, cluster, i)
+				if err == nil {
+					errs <- nil
+					return
+				}
+				time.Sleep(100 * time.Millisecond)
 			}
-			cancel()
-		}
+			if err == nil {
+				err = ctx.Err()
+			}
+			errs <- err
+		}(i)
+	}
 
-		if sealed >= n {
-			return nil
+	var merr *multierror.Error
+	var sealed int
+	for range cluster.Nodes() {
+		err := <-errs
```

Comment:
Would this need a poll to make sure it does not block if something unusual happens?

Reply:
Debatable.  My reasoning for why it doesn't: ctx ensures that every goroutine will terminate its for loop eventually, so every goroutine will write a value to the channel.  Ultimately I'd want any polling to also be constrained by ctx, which I think comes down to the same thing.  But then I'd also need another mechanism to figure out what to do about any goroutines that haven't yet written a value, so this seemed simpler.

Let me know if you see something I'm missing, happy to revisit it.

Reply:
Makes sense. Thanks for the context!